datalad/metadata/search.py

```diff
@@ -135,7 +135,7 @@ def _deep_kv(basekey, dct):
                 # temp files during index staging, and trips over absent "directories"
                 # TODO maybe even kill parentheses
                 # TODO actually, it might be better to have an explicit whitelist
-                k.replace(os.sep, '_').replace(' ', '_').replace('-', '_').replace('.', '_').replace(':', '-')
+                k.lstrip('@').replace(os.sep, '_').replace(' ', '_').replace('-', '_').replace('.', '_').replace(':', '-')
```

Comment:
it grows and might grow more... IMHO we should get a helper to make it more readable, e.g. the one in https://stackoverflow.com/a/2484247 which would have called `replace_many` or alike
```python
substitutions = [
    ('<', '&lt;'),
    ('>', '&gt;'),
    ...]

for search, replacement in substitutions:
    string = string.replace(search, replacement)
```
since apparently regular `.replace` even if chained still wins in speed over more sophisticated but explicit replacement solutions in that post

Reply:
I would not oppose such move.